### PR TITLE
Support extended-sys-state on Plane

### PR DESCRIPTION
**Pull request by peterbarker**
This was part of https://github.com/ArduPilot/ardupilot/pull/8587 - but since this isn't a trivial change like the Copter one I thought I'd split it off

### CODE REVIEW

File: Tools/autotest/quadplane.py [hunk 1]
```diff
@@ -155,6 +157,95 @@ def fly_mission(self, filename, fence, height_accuracy=-1):
...
+            if self.get_sim_time() - tstart > 10:
+                raise NotAchievedException("Did not achieve vol/landed states")
+            self.progress("Waiting for vtol_state=%u landed_state=%u" %
+                          (vtol_state, landed_state))
+            m = self.mav.recv_match(type='EXTENDED_SYS_STATE',
+                                    blocking=True,
+                                    timeout=1)
+            if m is None:
+                raise NotAchievedException("Did not get extended_sys_state message")
+            print(m)
```
Comment (OXINARF) [inline Tools/autotest/quadplane.py]: Do we really need to print the message?

**Comment by peterbarker** [comment]
On Tue, 5 Mar 2019, Francisco Ferreira wrote:

> +            print(m)
> 
> Do we really need to print the message?

Well, no - but it is useful to see what transitions you're seeing.  It's 
not high-rate and nothing like as noisy as a lot of our tests!

**Comment by peterbarker** [comment]
I've popped off the changes in this PR which actually affected the behaviour of the vehicle outside sending the mavlink messages.

I think that should make this non-controversial.  I'll open another PR trying to make these more accurate in the future.

**Review by khancyr — APPROVED** [review]
This looks fine. 
We could complete with the landing and takeoff report later. I would said, let's merge this before more conflict arrived

**Comment by peterbarker** [comment]
QGC uses this information to augment its interface.

**Comment by IamPete1** [comment]
I have not tested, just written in a comment, but I think this is the correct logic for tailsitters. 

```
if (plane.quadplane.tailsitter.enabled()) {
    if (plane.quadplane.tailsitter.in_vtol_transition()) {
        return MAV_VTOL_STATE_TRANSITION_TO_MC;
    } elseif (!plane.quadplane.in_vtol_mode()) {
        if (plane.quadplane.tailsitter.is_in_fw_flight()) {
            return MAV_VTOL_STATE_FW;
        } else {
            return MAV_VTOL_STATE_TRANSITION_TO_FW;
        }
    }
    return MAV_VTOL_STATE_MC;
}
```
This is a little clunky in that all of the tailsitter functions check tailsitter enable again and such, but the re-work would make it more opaque.

**Pull request closed without merging**